pandas/__init__.py

```diff
@@ -202,8 +202,40 @@
         FutureWarning,
         stacklevel=2,
     )
-# Don't allow users to use pandas.os or pandas.warnings
-del os, warnings
+
+# DeprecationWarning for missing pyarrow
+from pandas.compat.pyarrow import pa_version_under10p1
+
+if pa_version_under10p1:
+    # Check if old pyarrow is installed
+    from pandas.compat._optional import VERSIONS
+
+    try:
+        import pyarrow
+
+        pa_msg = (
+            f"was too old on your system - pyarrow {VERSIONS['pyarrow']} "
+            "is the current minimum supported version as of this release."
+        )
+        del pyarrow
+    except ImportError:
+        pa_msg = "was not found to be installed on your system."
+
+    warnings.warn(
+        f"""
+Pyarrow will become a required dependency of pandas in the next major release of pandas,
+(to allow more performant data types and better interoperability with other libraries)
+but {pa_msg}
+If this would cause problems for you,
+please provide us feedback at https://github.com/pandas-dev/pandas/issues/54466
+        """,
+        DeprecationWarning,
+        stacklevel=2,
```

Comment:
Do we want to make this stacklevel=1? 

Because this code is top-level and not inside a function or method, it one less than the typical stacklevel of 2 we use in a direct function/method. 
Setting it as 1 would avoid that it triggers a warning when you call a function that has an inline `import pandas` (not sure how important this is, though)

Reply:
The warning doesn't pop up in the REPL when I import pandas if I do stacklevel=1.

Reply:
Hmm, OK forget this. I did test it, but should have done something wrong because now I can't reproduce getting the warning with plain import and stacklevel 1